docs/repository.rst

```diff
@@ -0,0 +1,21 @@
+PyTeal Repository Structure
+===========================
+
+The main PyTeal codebase is logically organized as follows:
+
+
+pyteal/ast
+~~~~~~~~~~
+
+Code for the basic PyTeal building blocks: expressions which define an Abstract Syntax Tree (AST) .
+
+
+pyteal/compiler
+~~~~~~~~~~~~~~~
+
+The basic compilation algorithms.
+
+pyteal/ir
+~~~~~~~~~
+
+Code for PyTeal's Intermediate Representation (IR) which the compiler generates midway from the AST before finally converting to TEAL.
```

Comment:
Optional and mostly unrelated to the line:  

I think reviewing merged PRs for a particular class of problem is an effective way (for me) to accelerate learning.

In PyTeal's case, I think linking to a merged opcode addition PR is a great way to help contributors understand the space.
* Obviously, some opcodes are more involved than others.
* I think we'd want to pick an example that illustrates a complete definition of done (e.g. includes user guide doc updates).

If we think it's a productive idea and we don't have a candidate PR, we can always use the next opcode addition as the candidate.

Reply:
👍 - good addition